Support for Certificate Subject

ChrisLGardner opened the pull request:
Adding FarmCertificateSubject as a parameter that can be provided instead of a thumbprint. There are use cases where generating a certificate as part of the same DSC configuration was possible (and potentially likely) so the thumbprint isn't known. 

The actual method used here is pretty simple but seems to work pretty well in my testing. Obviously requires that DependsOn is used to allow the certificate request to happen first.

Commit:
Adding handling passing in a cert subject.

If a cert is generated at the same time as ServiceBus is configured then the thumbprint is unknown. Passing in the cert subject (which will be known) means we can get the thumbprint and pass that on to New-SBFarm as normal.

Commit:
Adding parameter for CertificateSubject

Commit:
Updating readme for FarmCertificateSubject

ryanspletzer commented:
Thanks for the contribution, I need to look it over, I think some tests need to be adjusted for the build to pass... But this makes sense to me. :)

ChrisLGardner commented:
I will have a look at those and also add in an example and add them into this PR.

Commit:
Fixing errors with the tests.

Shortened the line that was added in the readme.

Moved where we remove the FarmCertificateSubject from the NewSBFarm method and added a line to remove it from SetSBFarm method.

Commit:
Removed trailing space.

Commit:
Removed length validation, due to copy/paste mistake

Commit:
Fixing test to account for new parameter

ryanspletzer commented:
Awesome, thanks!
